homework/index.js

```diff
@@ -29,23 +31,154 @@
-        });
-        return;
+  function createTableRow(table, header, optionsValue = {}) {
+    const tr = createAndAppend('tr', table);
+    createAndAppend('td', tr, { text: header, class: 'td-header' });
+    createAndAppend('td', tr, optionsValue);
+    return tr;
+  }
+
+  function formatDate(dateString) {
+    const dateTime = new Date(dateString);
+    return dateTime.toLocaleString();
+  }
+
+  function renderRepoDetails(repo, section) {
+    section.innerHTML = '';
+    const table = createAndAppend('table', section);
+    let description;
+    if (repo.description) {
+      description = repo.description;
+    } else {
+      description = 'No description';
+    }
+    const firstRow = createTableRow(table, 'Repository:');
+    createAndAppend('a', firstRow.lastChild, {
+      text: repo.name,
+      href: repo.html_url,
+      target: '_blank',
+    });
+
+    createTableRow(table, 'Description:', { text: description });
+    createTableRow(table, 'Forks:', { text: repo.forks });
+    createTableRow(table, 'Updated', { text: formatDate(repo.updated_at) });
+  }
+
+  function createListItem(ul, contributor) {
+    const li = createAndAppend('li', ul, { class: 'list-item' });
+    createAndAppend('img', li, {
+      src: contributor.avatar_url,
+      alt: contributor.login,
+      class: 'contributor-img',
+    });
+    createAndAppend('a', li, {
+      text: contributor.login,
+      href: contributor.html_url,
+      target: '_blank',
+    });
+    createAndAppend('div', li, {
+      text: contributor.contributions,
+      class: 'contributions-div',
+    });
+  }
+
+  function renderContribution(contributorsUrl, section) {
+    section.innerHTML = '';
+    const ul = createAndAppend('ul', section);
+    createAndAppend('p', ul, { text: 'Contributions' });
+
+    const contributorsPromise = fetchJSON(contributorsUrl);
+    contributorsPromise
+      .then(data => {
+        data.forEach(contributor => createListItem(ul, contributor));
+      })
+      .catch(err => {
+        createErrorMessage(err);
+      });
+  }
+
+  function selectRepository(select, mainDiv, input, initialSelectValue = '') {
+    return function() {
+      mainDiv.innerHTML = '';
+      const basicInfoSec = createAndAppend('section', mainDiv);
+      const contributionsSec = createAndAppend('section', mainDiv, {
+        class: 'contributions-section',
+      });
+      const selectedRepoIndex =
+        select.options[select.options.selectedIndex].value;
+
+      if (initialSelectValue === 0) {
+        renderRepoDetails(input, basicInfoSec);
+        renderContribution(input.contributors_url, contributionsSec);
+      } else {
+        renderRepoDetails(input[selectedRepoIndex], basicInfoSec);
+        renderContribution(
+          input[selectedRepoIndex].contributors_url,
+          contributionsSec,
+        );
       }
-      const ul = createAndAppend('ul', root);
-      repos.forEach(repo => renderRepoDetails(repo, ul));
+    };
+  }
```

Comment:
selectRepository function was the most difficult part of my homework because I couldn't figure out how I should display the first option  when the page first loaded. Now that I see your feedback, I am thinking the logic was very simple actually. The first option is already selected so I could have build my whole logic accordingly. I am really glad to receive such detailed feedback since it really made things much more clear to understand on my part. Thank you very much.